crates/pathfinder/src/state/sync.rs

```diff
@@ -302,6 +354,28 @@ where
 
                     tracing::trace!("Query for existence of contracts: {:?}", contracts);
                 }
+                Some(l2::Event::Pending(block, state_update)) => {
+                    let latest_root = tokio::task::block_in_place(|| {
+                            let tx = db_conn.transaction()?;
+                            StarknetBlocksTable::get(&tx, StarknetBlocksBlockId::Latest)
+                        }
+                    )
+                    .context("Query latest state root for pending update")?
+                    .map(|block| block.root)
+                    .unwrap_or(GlobalRoot(StarkHash::ZERO));
+
+                    if state_update.old_root != latest_root {
+                        tracing::error!(latest=%latest_root, pending=%state_update.old_root,
+                            "Pending old state root does not match latest state root"
```

Comment:
Do you have an idea for how could this situation happen? Could comment it. I cannot really see because it would seem that the pending poller filters already.

Reply:
This can happen if there was a new block or reorg between the pending block fetch and this fetch.

Reply:
Ah I added it here because the polling doesn't check the root. But it probably should, because that could happen as described above.

Reply:
I think it would make sense to comment why we suspect this might happen.